### PR TITLE
Standardize exception handling for login

### PR DESCRIPTION
**Pull request by godfreyyeung**
Will properly display error for the /login route, like so:
image

Also document the need for logging out errors

### CODE REVIEW

File: server/src/app.controller.ts [hunk 1]
```diff
@@ -29,11 +30,15 @@ export class AppController {
...
-        res.status(401).send({ errors: [e] });
+        throw e;
       } else {
-        console.log(e);
-
-        res.status(500).send({ errors: [e] });
+        const error = {
+          code: "LOGIN_ERROR",
+          title: "Failed to login",
+          detail: "An unknown error occured while authenticating user",
```
Comment (trbmcginnis) [inline server/src/app.controller.ts:38]: So this is like the "catch all" net that catches anything that slipped through the service correct?

File: server/src/contact/contact.service.ts [hunk 2]
```diff
@@ -35,9 +35,13 @@ export class ContactService {
 
       return firstRecord;
     } catch(e) {
-      const errorMessage = `Error finding contact by ID. ${e.message}`;
+      const errorMessage = `Error finding contact by ID, possibly due to missing or bad ID. ${e.message}`;
       console.log(errorMessage);
-      throw new HttpException(errorMessage, HttpStatus.UNAUTHORIZED);
+      throw new HttpException({
```
Comment (trbmcginnis) [inline server/src/contact/contact.service.ts:40]: Why is this formatted differently from the ones in `auth.service.ts`?

File: server/src/app.controller.ts (hunk 1 above)
Comment (godfreyyeung) [reply server/src/app.controller.ts:38]: yep, exactly what I was thinking!

File: server/src/contact/contact.service.ts (hunk 2 above)
Comment (godfreyyeung) [reply server/src/contact/contact.service.ts:40]: Ah, good catch. I should format it the same as auth.service.ts!

Comment (godfreyyeung) [reply server/src/contact/contact.service.ts]: Fixed! Thanks!

File: server/EXCEPTIONS.md [hunk 3]
```diff
@@ -1,7 +1,23 @@
...
+public async getProject(){
+  try {
+    return myProject = this.get('dcp_project', ...);
+  } catch (e) {
+    const error = {
+      code: 'NO_PROJECT_FOR_ID',
+      title: 'No project for id',
+      detail: `Failed to find a project for given id. ${e.message}`,
+    };
+    console.log(error); // for backend log records
```
Comment (godfreyyeung) [inline server/EXCEPTIONS.md:15]: Can we use out of the box HttpException logging features?

**Pull request merged**